flink-connector-jdbc/src/main/java/org/apache/flink/connector/jdbc/JdbcConnectionOptions.java

```diff
@@ -90,13 +98,31 @@ public JdbcConnectionOptionsBuilder withDriverName(String driverName) {
             return this;
         }
 
+        public JdbcConnectionOptionsBuilder withProperties(Properties properties) {
```

Comment:
I have mixed feelings about this method...

I understand that it may be practical, but as a user I can expect that the properties I am passing will override all properties already defined.

as an example:
```
.withProperty("a", "a1")
.withProperties(new properties())
.withProperty("b", "b1")
```

If you saw that (without knowing the implementation), what will you expect.
that a and b should be in the properties, or just b...

Reply:
hi, @eskabetxe Thanks a lot for your quick-review 😄 . 

```
.withProperty("a", "a1")
.withProperties(new properties())
.withProperty("b", "b1")
``` 

> .....
> that a and b should be in the properties, or just b...
It's right.
It's difficult to express , which depends the what cases devs need and I have to admit that the lines referenced above is ambiguous now.  

I have come up with two temporary ways to alleviate this ambiguity:

- Keep the current logic, and then we add comment to describe semantics for these two methods.

OR

- Add a flag parameter to method `withProperties(new properties(), true/false)` , indicating whether to force assign   entire `properties`, rather than a part of the `properties` by `upsert` mode.

Of course, I'd appreciated to hear any ideas form you / community devs ! 😃

Reply:
@RocMarshal I would say the easiest way, eliminating the method.

If someone has a properties, they can do:
`properties.forEach((key, value) -> builder.withProperty(key, value));`

Under the hood, the `properties.putAll` is what it does.

Reply:
Thanks for the comments.
That sounds nice to me 
updated~